ydb/core/mind/bscontroller/console_interaction.cpp

```diff
@@ -0,0 +1,209 @@
+        NTabletPipe::SendData(Self.SelfId(), *ConsolePipe, commitConfigEv.release());
+    }
+
+    void TConsoleInteraction::Handle(TEvBlobStorage::TEvControllerReplaceConfigRequest::TPtr &ev) {
+        STLOG(PRI_DEBUG, BS_CONTROLLER, BSC24, "Console replace config request", (Request, ev->Get()->Record));
+        auto& record = ev->Get()->Record;
+        auto response = MakeHolder<TEvBlobStorage::TEvControllerReplaceConfigResponse>();
+        if (!record.GetOverwriteFlag()) {
+            if (GRPCSenderId) {
+                response->Record.SetStatus(NKikimrBlobStorage::TEvControllerReplaceConfigResponse::OngoingCommit);
+                Self.Send(ev->Sender, response.Release());
+                return;
+            }
+        }
+        GRPCSenderId = ev->Sender;
+        if (!record.GetSkipConsoleValidation()) {
+            ValidationTimeout = false;
+            TActivationContext::Schedule(TDuration::Minutes(2), new IEventHandle(Self.SelfId(), Self.SelfId(), new TEvBlobStorage::TEvControllerValidationTimeout()));
+            auto validateConfigEv = std::make_unique<TEvBlobStorage::TEvControllerValidateConfigRequest>();
+            validateConfigEv->Record.SetYAML(Self.YamlConfig);
+            NTabletPipe::SendData(Self.SelfId(), *ConsolePipe, validateConfigEv.release());
+        }
+    }
+
+    void TConsoleInteraction::Handle(TEvBlobStorage::TEvControllerValidationTimeout::TPtr &ev) {
+        ValidationTimeout = true;
+        auto response = MakeHolder<TEvBlobStorage::TEvControllerReplaceConfigResponse>();
+        response->Record.SetStatus(NKikimrBlobStorage::TEvControllerReplaceConfigResponse::ValidationTimeout);
+        Self.Send(*GRPCSenderId, response.Release());
+    }
+
+    void TConsoleInteraction::Handle(TEvBlobStorage::TEvControllerValidateConfigResponse::TPtr &ev) {
+        STLOG(PRI_DEBUG, BS_CONTROLLER, BSC27, "Console validate config response", (Response, ev->Get()->Record));
+        auto& record = ev->Get()->Record;
+        auto status = record.GetStatus();
+        auto replaceConfigResponse = std::make_unique<TEvBlobStorage::TEvControllerReplaceConfigResponse>();
+        switch(status) {
+            case NKikimrBlobStorage::TEvControllerValidateConfigResponse::IdPipeServerMismatch:
+                MakeGetBlock();
+                MakeRetrySession();
+                return;
+            case NKikimrBlobStorage::TEvControllerValidateConfigResponse::ConfigNotValid:
+                replaceConfigResponse->Record.SetStatus(NKikimrBlobStorage::TEvControllerReplaceConfigResponse::ConsoleInvalidConfig);
+                Self.Send(*GRPCSenderId, replaceConfigResponse.release());
+                return;
+            case NKikimrBlobStorage::TEvControllerValidateConfigResponse::ConfigIsValid:
+                break;
+            default:
+                Y_FAIL("Unexpected console validate config response status: %d", status);
+        }
+        bool skipBSCValidation = record.GetSkipBSCValidation();
```

Comment:
why not just put it in argument a line below?
its meaning is still quite obvious